core/src/services/supabase/core.rs

```diff
@@ -0,0 +1,226 @@
+            self.endpoint,
+            self.bucket,
+            percent_encode_path(&p)
+        );
+
+        Request::get(&url)
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)
+    }
+
+    pub fn supabase_get_object_info_public_request(
+        &self,
+        path: &str,
+    ) -> Result<Request<AsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+        let url = format!(
+            "{}/storage/v1/object/info/public/{}/{}",
+            self.endpoint,
+            self.bucket,
+            percent_encode_path(&p)
+        );
+
+        Request::get(&url)
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)
+    }
+
+    pub fn supabase_get_object_info_auth_request(&self, path: &str) -> Result<Request<AsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+        let url = format!(
+            "{}/storage/v1/object/info/authenticated/{}/{}",
+            self.endpoint,
+            self.bucket,
+            percent_encode_path(&p)
+        );
+
+        Request::get(&url)
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)
+    }
+}
+
+// core utils
+impl SupabaseCore {
+    pub async fn send(&self, req: Request<AsyncBody>) -> Result<Response<IncomingAsyncBody>> {
+        self.http_client.send(req).await
+    }
+
+    pub async fn supabase_get_object(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        if self.key.is_some() {
+            self.supabase_get_object_auth(path).await
+        } else {
+            self.supabase_get_object_public(path).await
+        }
+    }
+
+    pub async fn supabase_get_object_info(
+        &self,
+        path: &str,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        if self.key.is_some() {
+            self.supabase_get_object_info_auth(path).await
+        } else {
+            self.supabase_get_object_info_public(path).await
+        }
+    }
+
+    async fn supabase_get_object_public(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.supabase_get_object_public_request(path)?;
+        self.sign(&mut req)?;
+        self.send(req).await
+    }
+
+    async fn supabase_get_object_auth(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.supabase_get_object_auth_request(path)?;
+        self.sign(&mut req)?;
+        self.send(req).await
+    }
+
+    async fn supabase_get_object_info_public(
+        &self,
+        path: &str,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.supabase_get_object_info_public_request(path)?;
+        self.sign(&mut req)?;
+        self.send(req).await
+    }
+
+    async fn supabase_get_object_info_auth(
+        &self,
+        path: &str,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.supabase_get_object_info_auth_request(path)?;
+        self.sign(&mut req)?;
+        self.send(req).await
```

Comment:
Please remove them all, we don't need to them.

Reply:
What do you mean by we don't need them, I do not quite understand. The `supabase_get_object` is used in `read` and `supabase_get_object_info` is used in `stat`

Reply:
I mean expand `supabase_get_object_public`, `supabase_get_object_auth`, `supabase_get_object` directly:

```rust
pub fn supabase_get_object(&self, path: &str) -> Result<Response<AsyncBody>> {
    let p = build_abs_path(&self.root, path);

    let req = if self.core.key {
       xxx
    } else {
       yyy
    };

    self.send(req).await
}
```